Sources/MapboxMaps/Snapshot/Snapshotter.swift

```diff
@@ -86,134 +89,131 @@ public class Snapshotter: Observer {
                 let mbxImage = try! snapshot.image()
-                let scale = UIScreen.main.scale
-
-                if let uiImage = UIImage(mbxImage: mbxImage, scale: scale) {
-                    let rect = CGRect(origin: .zero, size: uiImage.size)
-                    let renderer = UIGraphicsImageRenderer(size: uiImage.size)
-                    let compositeImage = renderer.image { rendererContext in
-
-                        // First draw the snaphot image into the context
-                        let context = rendererContext.cgContext
-
-                        if let cgImage = uiImage.cgImage {
-                            context.draw(cgImage, in: rect)
-                        }
-
-                        let pointForCoordinate = { (coordinate: CLLocationCoordinate2D) -> CGPoint in
-                            let screenCoordinate = try! snapshot.screenCoordinate(for: coordinate)
-                            return CGPoint(x: screenCoordinate.x, y: screenCoordinate.y)
-                        }
-
-                        let coordinateForPoint = { (point: CGPoint) -> CLLocationCoordinate2D in
-                            // TODO: Fix circular dependency issues with MapboxMapsStyle/Foundation in order to use point.screenCoordinate extension
-                            let screenCoordinate = ScreenCoordinate(x: Double(point.x), y: Double(point.y))
-                            return try! snapshot.coordinate(for: screenCoordinate)
-                        }
-
-                        // Apply the overlay, if provided.
-                        let overlay = SnapshotOverlay(context: context,
-                                                      scale: scale,
-                                                      pointForCoordinate: pointForCoordinate,
-                                                      coordinateForPoint: coordinateForPoint)
-
-                        if let overlayHandler = overlayHandler {
-                            context.saveGState()
-                            overlayHandler(overlay)
-                            context.restoreGState()
-                        }
-
-                        // Composite the logo on the snapshot,
-                        // only after everything else has been drawn.
-                        let logoView = MapboxLogoView(logoSize: .regular)
-                        let logoPadding = CGFloat(10.0)
-                        let logoOrigin = CGPoint(x: logoPadding,
-                                                 y: uiImage.size.height - logoView.frame.size.height - logoPadding)
-                        context.translateBy(x: logoOrigin.x, y: logoOrigin.y)
-                        logoView.layer.render(in: context)
-                     }
-                    completion(.success(compositeImage))
-                } else {
+                let scale = CGFloat(self.mapSnapshotOptions.pixelRatio)
+                guard let uiImage = UIImage(mbxImage: mbxImage, scale: scale) else {
```

Comment:
What could cause this to fail?

Reply:
(doesn't need to be addressed as part of this PR — I see this is just a refactor from `if` to `guard`, which I do like.